### ISSUE TITLE
[Bug] precision difference with rwkv7

### ISSUE THREAD

**Issue by SmerkyG** [fla-org/flash-linear-attention#181]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

I found something interesting... I think there's a missing cast to float() in FLA RWKV7 HF code

`w = -math.exp(-0.5) * self.w_lora(xw).sigmoid()`
is a great simplification of Blink's formula
`w = -F.softplus(-(self.w0 + torch.tanh(xw @ self.w1) @ self.w2)) - 0.5 # soft-clamp to (-inf, -0.5)`

BUT

Blink later calls `-exp(w)` to get log_w, which implicitly upcasts to float32 in automatic mixed precision
and simply calling tanh does NOT cast to float32 in AMP
this is an important precision difference in my experience 
simply casting to float would fix this, like:
`log_w = -math.exp(-0.5) * self.w_lora(xw).to(torch.float32).sigmoid() `

AMP examples of where upcasting takes place and where it does not:
```
>>> x=torch.ones(1, dtype=torch.bfloat16, device=0)
>>> with torch.autocast("cuda"): torch.exp(x).dtype
torch.float32
>>> with torch.autocast("cuda"): torch.sigmoid(x).dtype
torch.bfloat16
>>> with torch.autocast("cuda"): torch.tanh(x).dtype
torch.bfloat16
```

### Steps to Reproduce the Bug

long ctx

### Expected Behavior

should be possible to see better results on long context training and inference with this fix

### Environment Information

1. Torch:
2. Triton:

**Comment by yzhangcs** [comment]
Fixed by https://github.com/fla-org/flash-linear-attention/commit/965c355397641a0b75f2edd36789aa2f83e9961a
